### PR TITLE
Adds project.json/project.json.template files under external to auto-update list

### PR DESCRIPTION
**Pull request by weshaggard**
Fixes https://github.com/dotnet/corefx/issues/15219

cc @stephentoub @dagood

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test please (Jenkins failure, FYI @mmitche https://ci.dot.net/job/dotnet_corefx/job/master/job/osx_release_prtest/2100/consoleText)

**Comment by mmitche** [comment]
@stephentoub Thanks.  These are fixed in the newest version.

**Comment by weshaggard** [comment]
>Thanks for fixing this, @weshaggard. So now when there are auto-updates, we should expect to see all three of these files modified?

Yes it should be a set of these 3 files not necessarily all 3 each time. I got the project.json files updated by running the normal update command so the auto-updates should work after this is merged.

File: external/coreclr/project.json.template [hunk 1]
```diff
@@ -2,16 +2,16 @@
   "frameworks": {
     "{TFM}": {
       "dependencies": {
-        "Microsoft.NETCore.Platforms": "1.2.0-beta-24721-02",
-        "Microsoft.NETCore.Runtime.CoreCLR": "1.2.0-beta-24904-03",
-        "Microsoft.NETCore.TestHost": "1.2.0-beta-24904-03",
-        "runtime.native.System.Data.SqlClient.sni": "4.4.0-beta-24814-01",
+        "Microsoft.NETCore.Platforms": "2.0.0-beta-24916-02",
```
Comment (weshaggard) [inline external/coreclr/project.json.template]: @ericstj Is there any way we can pull in the runtime.json directly and let nuget see it without needing to have a dependency on this package which is produced in corefx?

Comment (ericstj) [inline external/coreclr/project.json.template]: We could easily do that but it won't work.  We have no way of handling the conflict between old and new runtime.jsons.  As a result NuGet would see two sets of the RID graph when packages brought in the old package.

Comment (weshaggard) [inline external/coreclr/project.json.template]: For the corefx build I want to remove the Microsoft.NETCore.Platform package so it shouldn't have any impact on our external deps.

Comment (ericstj) [inline external/coreclr/project.json.template]: I see, misread your original question.  I thought you were talking about eliminating the MS.NETCore.Platforms package.  
NuGet only picks up runtime.json if it appears in the root of a referenced package.  If you wanted to get rid of the consumption of this package you'd need to construct a package with this runtime.json in it and ensure that's part of the compile graph.

Comment (weshaggard) [inline external/coreclr/project.json.template]: Are there any other options? Perhaps a simple ProjectReference to the pkgproj will be enough (assuming it doesn't require any other things to be present given that the external directory builds early).

**Pull request merged**